docs/web-apps/automated-testing/selenium/pre-run-executables.md

```diff
@@ -307,3 +305,384 @@ If using Sauce Storage for your pre-run executable send the following desired ca
 ### 64 v. 32-bit Version of AutoIT
 
 The 64bit version of AutoIT works on IE11, and not on IE9. The 32bit version works with both browser versions.
+
+## Automated Testing
+
+### Upload Files with the REST API
+
+Below are some examples of how to use the Sauce Labs REST API to upload your mobile file to Sauce Storage. If you do not have a file to test, consider using the [Sauce Labs Swag Labs sample app](https://github.com/saucelabs/sample-app-mobile) for validating this process.
+
+#### REST API Authentication
+
+The authorization credentials are located here: [app.saucelabs.com](https://app.saucelabs.com/user-settings). A recommended best practice is to set your credentials as environment variables like so:
+
+```bash
+SAUCE_USERNAME='valid.username'
+SAUCE_ACCESS_KEY='valid.key'
+```
+
+For specific instructions on how to set environment variables visit, the following links:
+
+- [Set Environment Variables with Windows 10](https://www.architectryan.com/2018/08/31/how-to-change-environment-variables-on-windows-10/)
+- [Set Environment Variables with MacOS](https://apple.stackexchange.com/questions/106778/how-do-i-set-environment-variables-on-os-x)
+- [Set Environment Variables with Linux](https://askubuntu.com/questions/58814/how-do-i-add-environment-variables)
+
+#### Accepted File Types
+
+App Storage recognizes .apk files as Android apps and .ipa files as iOS apps, and will also parse a .zip file to determine whether a valid .app bundle exists, accepting it as an iOS app if it does. You can also upload and store other file types for generic use, such as a pre-run executable, package, or binary. Some of the formats for this type of use case include:
```

Comment:
```suggestion
App Storage identifies .apk files as Android apps and .ipa files as iOS apps. Additionally, it can analyze a .zip file to check for a valid .app bundle, which it will accept as an iOS app if found. You can also upload and store other file types for generic use, such as a pre-run executable, package, or binary. Some of the formats for this type of use case are as follows:
```